Bio/SeqUtils/CodonUsage.py

```diff
@@ -175,3 +179,199 @@ def print_index(self):
+       Sharp and Li (1987). It uses a reference set of highly-expressed genes
+       provided by the user, and computes the alignment of any candidate coding
+       sequence with the codon usage patterns seen in that reference set.
+       A known problem of CAI is that it computes the weight (index) for each
+       codon as the ratio between the frequency of that codon in the reference 
+       set and the largest frequency among its synonymous codons. This 
+       implicitly assumes that the background nucleotide distribution is 
+       uniform. On mutationally-biased genomes (such as those found in many 
+       bacterial clades), this assumption will backfire, because CAI will
+       attribute to translational selection the patterns of genome-wide 
+       mutational bias observed in the reference set (i.e. a weakly used codon
+       will be overvalued by CAI if it is aligned with the overall GC% content
+       of the reference set, and vice versa). nRCA removes this bias by 
+       computing each codon index as the ration between the observed codon
+       frequency in the reference set and its expected frequency (inferred as
+       the product of the positional frequencies of each the codon bases).
+       This has been shown to improve the correlation of nRCA with gene 
+       expression values with respect to CAI (Fox and Erill, DNA Research, 
+       17:3, 185-196, 2010).
+       Based on the BioPython Module Bio.SeqUtils.CodonUsage code for CAI.
+    """
+    
+    def __init__(self): 
+        """Initializes the class""" 
+        self.index = {} 
+        self.codon_count = {} 
+        self.first_pos_count = {}
+        self.second_pos_count = {}
+        self.third_pos_count = {}
+    
+    # use this method with predefined CAI index 
+    def set_nrca_index(self, index): 
+        """Set up an index to be used when calculating nRCA for a gene. 
+            Just pass a dictionary similar to the ErillLabEcoliIndex in the 
+            CodonUsageIndices module. 
+        """ 
+        self.index = index 
+    
+    def generate_index(self, fasta_file): 
+        """Generate a codon usage index from a FASTA file of CDS sequences. 
+           Takes a location of a FASTA file containing CDS sequences 
+           (which must all have a whole number of codons) and generates the 
+           nRCA codon usage index. 
+        """ 
+    
+        # first make sure we're not overwriting an existing index: 
+        if self.index != {} or self.codon_count != {}: 
+            raise ValueError("an index has already been set or a codon count " 
+                             "has been done. Cannot overwrite either.") 
+
+        # count codon occurrences in the file, as well as codon position base 
+        # counts
+        self._codon_count(fasta_file) 
+        
+        # compute total number of codons
+        total_codons = sum(self.codon_count.values()) + 0.0
```

Comment:
I assume the plus ``0.0`` is to turn this from an integer into a float, in order to avoid integer division under Python 2.

It might be cleaner to import the Python 3 style division?

```python
from __future__ import division
```